server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -325,6 +438,83 @@ public RecoverySettings(Settings settings, ClusterSettings clusterSettings) {
         );
     }
 
+    private void computeMaxBytesPerSec(Settings settings) {
+        validateBandwidthsSettings(settings);
+
+        // limit as computed before 8.1.0
+        final long defaultBytesPerSec = Math.max(INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.get(settings).getBytes(), 0L);
+
+        // available network bandwidth
+        final long networkBandwidthBytesPerSec = Math.max(availableNetworkBandwidth.getBytes(), 0L);
+
+        // available disk bandwidth
+        final long diskBandwidthBytesPerSec;
+        if (availableDiskReadBandwidth.getBytes() > 0L && availableDiskWriteBandwidth.getBytes() > 0L) {
+            final long diskReadBandwidth = Math.round(
+                availableDiskReadBandwidth.getBytes() * MAX_BYTES_PER_SEC_USER_FACTOR_READS_SETTING.get(settings)
+            );
+            final long diskWriteBandwidth = Math.round(
+                availableDiskWriteBandwidth.getBytes() * MAX_BYTES_PER_SEC_USER_FACTOR_WRITES_SETTING.get(settings)
+            );
+            diskBandwidthBytesPerSec = Math.min(diskReadBandwidth, diskWriteBandwidth);
+        } else {
+            diskBandwidthBytesPerSec = 0L;
+        }
+
+        final long availableBytesPerSec = Math.round(
+            Math.min(diskBandwidthBytesPerSec, networkBandwidthBytesPerSec) * MAX_BYTES_PER_SEC_USER_FACTOR_SETTING.get(settings)
```

Comment:
I think applying the read/write factor _and then_ the overall factor might lead to surprise, given that we have other settings like `cluster.routing.allocation.node_concurrent{,_incoming,_outgoing}_recoveries` where the unqualified setting sets both of the qualified ones.

Reply:
Agreed, can we instead make `node.recovery.max_bytes_per_sec.user_factor.writes` default to the `default` one.

I think we should then make the calculation by calculating the read speed as `min(availableDiskReadBandwidth, networkBandwidthBytesPerSec) * read_factor` (likewise for write speed).

Reply:
Right, I agree it makes more sense like you both suggested. I pushed a change in 6f7c47d02be6ace15acfea30be6622b4ea82944b.

The max. overcommit is calculated from `Math.min(read, write) * overcommitFactor` (ie what would have been the limit if bandwidths were applied ajusted to the overcommit ratio we allow). Let me know if you want it calculated differently.